### ISSUE TITLE
Add `IAccountStateDelta.RemoveState()` to unset state after it's once set

### ISSUE THREAD

**Issue by dahlia** [planetarium/libplanet#1383]
<del datetime="2021-12-23T05:08:26Z">States have been intended to be nullable, as they can be empty.  I believe the nullability on that parameter was missing when I added `#nullable enable` to the source file.</del>

<ins datetime="2021-12-23T05:08:26Z">See also the recent comment: <https://github.com/planetarium/libplanet/issues/1383#issuecomment-1000046682>.</ins>

**Comment by dahlia** [comment]
I changed the title, because an assumption/semantics of states that every state can be unset by filling it with `null`[^1] had been an undocumented specification only I (@dahlia) have been aware of.  I leave some context for historical accuracy:

In early days, I (@dahlia) intended `IAccountStateDelta.SetState()` to take a `null` value to unset the state.  However, at that time *IAccountStateDelta.cs* did not enable null checker yet, and there has never been any mention about such behavior on docs.

Later, I [enabled null checker on *IAccountStateDelta.cs*][1] (context: #458), but I omitted give a question mark to type of `IAccountStateDelta.SetState()`'s second parameter by mistake.  As a result, the meaning of the parameter changed from nullable `IValue` to non-nullable `IValue`.  (The original description of this issue addressed this.)

When @moreal worked on MPT (`ITrie`, `MerkleTrie`, etc), he interpreted that states cannot be unset once after they are set.  It was natural, because there was no clue at all both in code and docs.  In the end, `ITrie.Set()` and so on are designed to disallow unset states and explicitly throw `ArgumentNullException` if `null` value is passed.  This behavior is documented at least.

Lib9c, Libplanet's virtually client code in production as of now, also never tries to unset states.  They instead fill states with `Bencodex.Types.Null.Value`.

To sum up, our implementation, doc, and client code all assume that states cannot be unset after being set once.  We could implement a feature to unset states in the future, it would be accomplished by adding a separate method like `IAccountStateDelta.RemoveState()`.  This issue would be left for that work.

[^1]: It differs `Bencodex.Types.Null.Value`.

[1]: https://github.com/planetarium/libplanet/commit/bb9b3f6bc876070365c890edbc4b89fb9b9418fe#diff-5eb3b56bb0dbcd80ad19fd7ce34e6fbd74d9694c6cd66f8eb68745a34510cd9f
